mlflow/sklearn/__init__.py

```diff
@@ -1421,65 +1425,127 @@ def patched_model_score(original, self, *args, **kwargs):
+            patched_IffHasAttrDescriptor__get__
+
+    def patch_estimator_method_if_available(class_def, func_name, patched_fn, manage_run):
+        if not hasattr(class_def, func_name):
+            return
+
+        original = getattr(class_def, func_name)
+        # Retrieve raw attribute while bypassing the descriptor protocol
+        raw_original_obj = gorilla.get_attribute(class_def, func_name)
+
+        if isinstance(raw_original_obj, property):
+            def real_original(self, *args, **kwargs):
+                bound_delegate_method = raw_original_obj.fget(self)
+                return bound_delegate_method(*args, **kwargs)
+
+            safe_patch(
+                FLAVOR_NAME, class_def, func_name, patched_fn, manage_run=manage_run,
+                original_fn=real_original
+            )
+            safe_patch_fn = getattr(class_def, func_name)
+
+            def get_bound_safe_patch_fn(self):
+                # This `raw_original_obj.fget` call is for availability check, if it raise error
+                # then `hasattr(obj, {func_name})` will return False
+                # so it mimic the original property behavior.
+                raw_original_obj.fget(self)
+
+                def bound_safe_patch_fn(*args, **kwargs):
+                    return safe_patch_fn(self, *args, **kwargs)
 
-        for func_name in ["predict", "predict_proba", "transform"]:
-            if should_patch_estimator_method(class_def, func_name):
-                safe_patch(
-                    FLAVOR_NAME, class_def, func_name, patched_predict, manage_run=False,
+                bound_safe_patch_fn = update_wrapper_extended(
+                    bound_safe_patch_fn, raw_original_obj.fget
                 )
+                return bound_safe_patch_fn
 
-        if should_patch_estimator_method(class_def, "score"):
+            get_bound_safe_patch_fn = update_wrapper_extended(
+                get_bound_safe_patch_fn, raw_original_obj.fget
+            )
+            setattr(class_def, func_name, property(get_bound_safe_patch_fn))
+
+        elif raw_original_obj is original and callable(original):
+            # normal method
+            safe_patch(
+                FLAVOR_NAME, class_def, func_name, patched_fn, manage_run=manage_run,
+            )
+        elif hasattr(raw_original_obj, 'delegate_names') or hasattr(raw_original_obj, 'check'):
             safe_patch(
-                FLAVOR_NAME, class_def, "score", patched_model_score, manage_run=False,
+                FLAVOR_NAME, class_def, func_name, patched_fn, manage_run=manage_run,
+            )
+            safe_patch_fn = getattr(class_def, func_name)
+
+            if hasattr(raw_original_obj, 'delegate_names'):
+                # the method is decorated by `@if_delegate_has_method`
+                decorated_safe_patch_fn = raw_original_obj.__class__(
+                    safe_patch_fn, raw_original_obj.delegate_names, func_name
+                )
```

Comment:
Just to confirm, in scikit-learn >= 1.0, we don't need this block of code, correct?